gui/src/components/LayoutComponents/SideBar.tsx

```diff
@@ -24,34 +35,55 @@ const SideBar: React.FC = () => {
     [pathname],
   );
 
+  const filteredOptions = useMemo(() => {
+    const projectType = getProjectTypeFromFramework(projectFramework);
+    if (projectType === projectTypes.DESIGN) {
+      return sidebarOptions.filter((option) =>
+        ['design', 'workbench'].includes(option.id),
+      );
+    } else if (projectType === projectTypes.BACKEND) {
+      return sidebarOptions.filter((option) =>
+        ['board', 'workbench', 'code', 'pull_request'].includes(option.id),
+      );
+    }
+    return null;
+  }, [projectFramework]);
+
   const renderedOptions = useMemo(() => {
-    return sidebarOptions.map((option, index) => (
-      <div
-        key={index}
-        id={`${option.id}_${index}_option`}
-        className={`${
-          handleSelectedOption(option)
-            ? 'sidebar_item_selected'
-            : 'sidebar_item'
-        } flex flex-col items-center justify-center gap-1 text-white`}
-        onClick={() => handleOptionClick(option)}
-      >
-        <Image
-          width={24}
-          height={24}
-          sizes={'100vw'}
-          src={
-            handleSelectedOption(option) ? option.selected : option.unselected
-          }
-          alt={`${option.id}_icon`}
-          priority
-        />
-        <span className="proxima_nova text-[9px] font-normal">
-          {option.text}
-        </span>
-      </div>
-    ));
-  }, [handleOptionClick, handleSelectedOption]);
+    return (
+      filteredOptions &&
+      filteredOptions.map((option, index) => {
+        return (
+          <div
+            key={index}
+            id={`${option.id}_${index}_option`}
+            className={`${
+              handleSelectedOption(option)
+                ? 'sidebar_item_selected'
+                : 'sidebar_item'
+            } flex flex-col items-center justify-center gap-1 text-white`}
+            onClick={() => handleOptionClick(option)}
+          >
+            <Image
+              width={24}
+              height={24}
+              sizes={'100vw'}
```

Comment:
sizes is not necessary if you are passing proper values for width and height

Reply:
got it, resolved